datafusion/optimizer/Cargo.toml

```diff
@@ -56,5 +56,6 @@ regex-syntax = "0.8.0"
 [dev-dependencies]
 arrow-buffer = { workspace = true }
 ctor = { workspace = true }
+datafusion-functions-aggregate = { workspace = true }
```

Comment:
🤔  -- I thought we were trying to avoid making the optimizer depend on the function library -- can we use the sub implementation instead?

Reply:
🤔  -- I can't remember anymore. 

It is probably cleaner to to avoid the dependencies, but we can clean it up as a follow on PR